File: python and java/第k大数.py
#快排的思想，每一轮过后，下次都只再排另一边。
class Solution:
	def kthLargestElement(self,k,A):
		result = self.quickSort(A,0,len(A) - 1,k)
		return result

	def quickSort(self,A,low,high,k):
		if low > high:
			return
		key = A[low]
		p = low
		q = high
		while p < q:
			while A[q] <= key and p < q:
				q = q - 1
			A[p] = A[q]
			while A[p] >= key and p < q:
				p = p + 1
			A[q] = A[p]
		A[p] = key
		if p > k - 1:
			return self.quickSort(A,low,p - 1,k)
		elif p == k - 1:
			return A[p]
		else:
			return self.quickSort(A,p + 1,high,k)

solution = Solution()
print(solution.kthLargestElement(10,[1,2,3,4,5,6,8,9,7,10]))

File: python and java/第k大数.java
class Solution{

	public int kthLargestElement(int k , int[] nums){
		int result = quickSort(k,nums,0,nums.length - 1);
		return result;
	}

	public Integer quickSort(int k,int[] nums,int left,int right){
		if (left > right) {
			return null;
		}
		int p = left;
		int q = right;
		int key = nums[left];
		while (p < q) {
			while (p < q && nums[q] <= key) {
				q--;
			}
			nums[p] = nums[q];
			while(p < q && nums[p] >= key){
				p++;
			}
			nums[q] = nums[p];
		}
		nums[p] = key;	
		if (p < k - 1) {
			return quickSort(k,nums,p + 1,right);
		}
		else if (p == k - 1) {
			return nums[p];
		}
		else{
			return quickSort(k,nums,left,p - 1);
		}
	}
}; 

public class 第k大数 {
	public static void main(String[] args) {
		Solution solution = new Solution();
		int arry[] = {1,2,3,8,5,3};
		int s = solution.kthLargestElement(5,arry);
		System.out.println(s);
	}
}
